src/server/game/Chat/Chat.cpp

```
@@ -964,15 +1000,75 @@ ObjectGuid::LowType ChatHandler::extractLowGuidFromLink(char* text, HighGuid& gu
 std::string ChatHandler::extractPlayerNameFromLink(char* text)
 {
     // |color|Hplayer:name|h[name]|h|r
-    char* name_str = extractKeyFromLink(text, "Hplayer");
-    if (!name_str)
+    std::string name_str = extractKeyFromLink(text, "Hplayer");
+    if (name_str.empty())
         return "";
 
-    std::string name = name_str;
-    if (!normalizePlayerName(name))
+    if (!normalizePlayerName(name_str))
         return "";
 
-    return name;
+    return name_str;
+}
+
+std::string ChatHandler::extractPlayerNameFromLink(std::string const &text)
+{
+    // |color|Hplayer:name|h[name]|h|r
+    std::string name_str = extractKeyFromLink(text, "Hplayer");
+    normalizePlayerName(name_str);
```

Ignoring errors from `normalizePlayerName`